lib/OpenQA/Parser/Format/XUnit.pm

```diff
@@ -76,7 +79,8 @@ sub parse {
                 my $content = '# Test messages ';
                 $content .= "# $tc->{name}\n" if $tc->{name};
 
-                for my $out ($tc->children('skipped, passed, error, failure')->each) {
+                for my $out ($tc->children('skipped, passed, error, failure, softfailure')->each) {
+                    $tc_result = 'softfail' if ($out->tag =~ m/softfailure/);
                     $tc_result = 'fail' if ($out->tag =~ m/failure|error/);
```

Comment:
The tag `softfailure` contains `failure`. So won't this assign `fail` even if just a `softfailure` tag is present due to the lack of anchoring in this regex?

I would at some point avoid the repetitive use of `$out->tag` in this loop and store that value a variable. Then one could also use interpolation in the line below.